lib/spack/docs/tutorial_advanced_packaging.rst

```diff
@@ -0,0 +1,560 @@
+.. _advanced-packaging-tutorial:
+
+============================
+Advanced Topics in Packaging
+============================
+
+While you can quickly accomplish most of the mundane tasks with what
+was covered in :ref:`packaging-tutorial`, there are times when such
+knowledge won't suffice. Usually this happens for libraries that provide
+more than one API and need to let dependents decide which one to use,
+or for packages that provide tools that are invoked at build-time,
+or in other similar situations.
+
+In the following we'll dig into some of the details of package's
+implementation that permit to deal with these rare, but important,
+occurrences. You can rest assured that in every case Spack remains faithful to
+its philosophy: keep simple things simple, but be flexible enough when
+complex requests arise!
+
+-----------------------
+Set up for the tutorial
+-----------------------
+
+The preferred way to follow this tutorial is to use a Docker image:
```

Comment:
"The simplest way to follow along with this tutorial is to use our Docker image, which comes with Spack and various packages pre-installed"

Then combine the console blocks.